src/App.js

```diff
@@ -16,8 +17,43 @@ class App extends React.Component {
       phones: getAll(),
       selectedPhone: null,
       basketItems: [],
-    };
-  }
+}}
+
+    onPhoneSelected = (phoneId) => {
+                      this.setState({
+                        selectedPhone: getById(phoneId),
+                      });
+                    };
+
+    onBasket = (phoneId) => {
+     this.setState((prevState) => { 
+                                    let dubletItem = prevState.basketItems.findIndex((item) => {return item.name === phoneId});
```

Comment:
```suggestion
  const itemIndex = prevState.basketItems.findIndex((item) => item.name === phoneId);
```

- Don't use `let` for variables which are not changing. Use `const` instead.
- Variable name should be self explained. Suggested name: `itemIndex`
- Use arrow function's `return` feature: you don't need to write word `return` explicitly if you return value immediately after invocation

Reply:
Suggestion: move calculation logic out of `setState`. It will be easier to understand. Typically `setState` should have minimum calculations inside. It should only return new `state` object.
You can use `this.state` instead of `prevState` when you refactor the code.